python/llm/src/bigdl/llm/cmd/wrapper.py

```diff
@@ -0,0 +1,110 @@
+
+
+def llama(args):
+    if os.name == 'nt':
+        exec_path = os.path.join(LIB_DIR, 'main-llama.exe')
+    else:
+        exec_path = os.path.join(LIB_DIR, 'main-llama')
+    command = [exec_path,
+               '-t', args.threads,
+               '-p', args.prompt,
+               '-n', args.n_predict,
+               '-m', args.model]
+    subprocess.run(command, check=True)
+
+
+def bloomz(args):
+    if os.name == 'nt':
+        exec_path = os.path.join(LIB_DIR, 'main-bloomz.exe')
+    else:
+        exec_path = os.path.join(LIB_DIR, 'main-bloomz')
+    command = [exec_path,
+               '-t', args.threads,
+               '-p', args.prompt,
+               '-n', args.n_predict,
+               '-m', args.model]
+    subprocess.run(command, check=True)
+
+
+def redpajama(args):
+    if os.name == 'nt':
+        exec_path = os.path.join(LIB_DIR, 'main-gptneox.exe')
+    else:
+        exec_path = os.path.join(LIB_DIR, 'main-gptneox')
+    command = [exec_path,
+               '-t', args.threads,
+               '-p', args.prompt,
+               '-n', args.n_predict,
+               '-m', args.model]
+    subprocess.run(command, check=True)
+
+
+model_exec_map = {
+    'llama': llama,
+    'bloomz': bloomz,
+    'redpajama': redpajama
+}
+
+
+def get_model_family(args):
+    # TODO: get model family by model path content
+    global model_exec_map
+    model_basename = os.path.basename(args.model)
+    for model_family in model_exec_map.keys():
+        if model_family in model_basename:
+            return model_family
+    model_dirname = os.path.basename(os.path.dirname(args.model))
+    for model_family in model_exec_map.keys():
+        if model_family in model_dirname:
+            return model_family
+    return 'llama'
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Command line wrapper')
+
+    parser.add_argument('-t', '--threads', type=int, default=28,
+                        help='number of threads to use during computation (default: 28)')
+    parser.add_argument('-p', '--prompt', type=str, default='empty',
+                        help='prompt to start generation with (default: empty)')
+    parser.add_argument('-n', '--n_predict', type=int, default=-1,
+                        help='number of tokens to predict (default: -1, -1 = infinity)')
+    parser.add_argument('-m', '--model', type=str,
+                        default='llama.cpp/models/lamma-7B/ggml-model.bin',
+                        help='model path (default: llama.cpp/models/lamma-7B/ggml-model.bin)')
+    parser.add_argument('--model_family', type=str, default='',
+                        choices=['llama', 'bloomz', 'redpajama'],
+                        help='family name of model')
+
+    args = parser.parse_args()
+    args.model = os.path.abspath(args.model)
+
+    if args.model_family:
+        model_exec_map[args.model_family](args)
+    else:
```

Comment:
I don't think this makes sense - there is no accurate way to detect the model family.

Reply:
This is a temporary solution, the purpose is to deal with the situation when the user does not provide the model family. In the design, it is planned to identify the type of the model through the parameters of the model file, but this part has not been completed.

Reply:
I don't think we need to support that; we should simple remove it from this PR.